docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/home_landing/paragraph--ncids-promo-block-external.html.twig

```diff
@@ -0,0 +1,59 @@
+{# ###### Setup Image Position & Image ####### #}
+{% set image_position = content.field_image_position_promo|render|striptags|lower|trim %}
+{% if image_position == 'no-img' %}
+    {% set image_position = '' %}
+{% elseif image_position == 'left' %}
+    {% set image_position = 'nci-promo-block--with-image' %}
+    {% set image = content.field_override_image_promotional %}
+{% else %}
+    {% set image_position = 'nci-promo-block--with-image nci-alternating-right' %}
+    {% set image = content.field_override_image_promotional %}
+{% endif %}
+
+{# ###### Setup Theme ####### #}
+{% set promo_theme = content.field_ncids_theme|render|striptags|lower|trim %}
+{% if promo_theme == 'light' %}
+    {% set promo_theme = '' %}
+{% endif %}
+
+{# ###### Setup button ####### #}
+{% set externalUrl = content.field_featured_url[0]['#url']|render %}
+{% if content.field_button_alt|render|striptags|trim is not empty %}
```

Comment:
I would say the same about the button alt -- if you don't need it rendered, then access the node. Since this is just a simple text field you can do that. Also, if doing the if, and then a set later, set it first and then IF some condition then do something with it.  You are re-rendering multiple fields multiple times, then passing it off to string manipulation functions each time. This is just a waste of CPU cycles. Set it once, cleaning up there, and everywhere else you reference the field is just like a normal variable.

Reply:
FYI, if the button alt were a WYSIWYG, then the render would be necessary. Although with WYSIWYG, you don't want to strip the tags.